syncto/authentication.py

```diff
@@ -38,5 +40,18 @@ def build_sync_client(request):
 
     bid_assertion = authorization_header.split(" ", 1)[1]
     client_state = request.headers[CLIENT_STATE_HEADER]
-    sync_client = SyncClient(bid_assertion, client_state)
+    try:
+        sync_client = SyncClient(bid_assertion, client_state)
+    except HTTPError as e:
+        if e.response.status_code in (400, 401):
+            response = http_error(httpexceptions.HTTPUnauthorized(),
+                                  errno=ERRORS.INVALID_AUTH_TOKEN,
+                                  message='%s %s: %s' % (
+                                      e.response.status_code,
+                                      e.response.reason,
+                                      e.response.text))
+            response.headers.extend(forget(request))
```

Comment:
Please add a comment here stating that we want to forget the set of credentials possessed by the currently connected user. I had to read [pyramid doc](http://docs.pylonsproject.org/projects/pyramid/en/latest/api/security.html#pyramid.security.forget) to understand what was actually going on here.